src/agoradatatools/etl/utils.py

```diff
@@ -299,3 +300,30 @@ def remove_duplicates_keep_order(lst: List[Any]) -> List[Any]:
         if item not in result:
             result.append(item)
     return result
+
+
+def convert_transformation_result_to_dataframe(
+    result: Union[DataFrame, Dict[str, Any], List[Dict[str, Any]], Any],
+    dataset_name: str,
+) -> Union[DataFrame, Dict[str, Any]]:
+    """Convert the result of a custom transformation to a pandas DataFrame.
+
+    Args:
+        result: The result from a custom transformation function
+        dataset_name (str): Name of the dataset for error reporting
+
+    Returns:
+        DataFrame: The converted result as a pandas DataFrame or a dictionary
```

Comment:
As above, edit the first "DataFrame:" to be "DataFrame or dict:" so that it's clear when glancing at it that this function can return either one.